### PR TITLE
fix(new-widget-builder-experience): Add queries to handleBlur dependency list

### LINKED ISSUE #33236: Error in editing/creating Widget on custom dashboard
### Environment

SaaS (https://sentry.io/)

### Version

_No response_

### Steps to Reproduce

1. Edit a widget by editing any of the queries or just by focusing and unfocusing query text input.

* First picture is when I first land on the edit widget page
* Second picture is when I just focus and un-focus the query text input in the `Filter your results` section.
* Third picture is when I edit the query, the query is still valid but then it shows an error.
* Fourth picture is when I try to submit it, it shows this error `conditions are required during creation.`

Screenshot 2022-04-02 at 12 20 20 PM
Screenshot 2022-04-02 at 12 21 13 PM
Screenshot 2022-04-02 at 12 19 56 PM
Screenshot 2022-04-02 at 12 42 05 PM

### Expected Result

It should show proper visualization for valid queries. This is a bug in the Sentry dashboard as it throws errors even when the query is not changed.

### Actual Result

It throws an error, as shown in the screenshots above.

### PR DESCRIPTION
**Pull request by narsaynorath**
The handleBlur has a dependency on queries, which was not updating
when the query length would change. This stale data meant that there
could be an indexing issue that malforms our query structure, leading to
a validation bug

The problem is that the `useCallback` hook doesn't update the `queries` list properly when it updates (either when the widget builder re-renders and fills with data, or when starting fresh and adding fields), leading to the `queries[queryIndex]` index to return undefined, which wipes out a bunch of expected fields. The solution is to pass `queries` to `useCallback` so it updates.

Fixes #33236

### CODE REVIEW

**Review by edwardgou-sentry — APPROVED** [review]
LGTM, also confirmed it fixes the undefined reference issue.

**Comment by priscilawebdev** [comment]
Thanks for fixing it @narsaynorath 🌮

File: static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx [hunk 1]
```diff
@@ -73,17 +73,20 @@ export function FilterResultsStep({
...
+        if (!blurTimeoutRef.current) {
+          const newQuery: WidgetQuery = {
+            ...queries[queryIndex],
+            conditions: field,
+          };
+          onQueryChange(queryIndex, newQuery);
+        }
+      };
+    },
+    [queries]
```
Comment (priscilawebdev) [inline static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx:88]: oh that was the queries... funny, I  would have fixed this in the PR https://github.com/getsentry/sentry/pull/33241

Comment (priscilawebdev) [reply static/app/views/dashboardsV2/widgetBuilder/buildSteps/filterResultsStep/index.tsx:88]: we have to do the same with `handleChange` 😉

**Pull request merged**